api/src/main/java/org/apache/iceberg/io/CloseableIterable.java

```diff
@@ -32,6 +32,31 @@
 import org.apache.iceberg.exceptions.RuntimeIOException;
 
 public interface CloseableIterable<T> extends Iterable<T>, Closeable {
+
+  default CloseableIterator<T> closeableIterator() {
```

Comment:
Why doesn't this update `iterator` to return a `CloseableIterator`? I don't think it is a good idea to have two methods that do basically the same thing, but one provides a closeable version. Then we need to remember to call this variant.

Reply:
Well,  sounds good to unify the closeableIterator and iterator methods into one, let me update the patch. Thanks.